pkg/component/kubeproxy/resources.go

```diff
@@ -654,3 +566,128 @@ func (k *kubeProxy) getMode() kubeproxyconfigv1alpha1.ProxyMode {
 	}
 	return "iptables"
 }
+
+func (k *kubeProxy) getInitContainers(kubernetesVersion *semver.Version, image string) []corev1.Container {
+	initContainers := []corev1.Container{
+		{
+			Name:            "cleanup",
+			Image:           image,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command: []string{
+				"sh",
+				"-c",
+				fmt.Sprintf("%s/%s %s", volumeMountPathCleanupScript, dataKeyCleanupScript, volumeMountPathMode),
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name:  "KUBE_PROXY_MODE",
+					Value: string(k.getMode()),
+				},
+				{
+					Name:  "EXECUTE_WORKAROUND_FOR_K8S_ISSUE_109286",
+					Value: strconv.FormatBool(versionutils.ConstraintK8sLess125.Check(kubernetesVersion)),
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				Privileged: pointer.Bool(true),
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      volumeNameCleanupScript,
+					MountPath: volumeMountPathCleanupScript,
+				},
+				{
+					Name:      volumeNameKernelModules,
+					MountPath: volumeMountPathKernelModules,
+				},
+				{
+					Name:      volumeNameDir,
+					MountPath: volumeMountPathDir,
+				},
+				{
+					Name:      volumeNameMode,
+					MountPath: volumeMountPathMode,
+				},
+				{
+					Name:      volumeNameKubeconfig,
+					MountPath: volumeMountPathKubeconfig,
+				},
+				{
+					Name:      volumeNameConfig,
+					MountPath: volumeMountPathConfig,
+				},
+			},
+		},
+	}
+
+	k8sGreaterEqual129 := versionutils.ConstraintK8sGreaterEqual129.Check(kubernetesVersion)
+	if k8sGreaterEqual129 {
+		initContainers = append(initContainers, k.getKubeProxyContainer(k8sGreaterEqual129, image, true))
+	}
+
+	return initContainers
+}
+
+func (k *kubeProxy) getKubeProxyContainer(k8sGreaterEqual129 bool, image string, init bool) corev1.Container {
+	container := corev1.Container{
+		Name:            containerName,
+		Image:           image,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command: []string{
+			"/usr/local/bin/kube-proxy",
+			fmt.Sprintf("--config=%s/%s", volumeMountPathConfig, dataKeyConfig),
+			"--v=2",
+		},
+		SecurityContext: &corev1.SecurityContext{
+			Capabilities: &corev1.Capabilities{
+				Add: []corev1.Capability{"NET_ADMIN"},
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("20m"),
+				corev1.ResourceMemory: resource.MustParse("64Mi"),
+			},
+		},
+		Ports: []corev1.ContainerPort{{
+			Name:          portNameMetrics,
+			ContainerPort: portMetrics,
+			HostPort:      portMetrics,
+			Protocol:      corev1.ProtocolTCP,
+		}},
```

Comment:
Additionally there is the following warning when creating the kube-proxy manifest manually with kubectl (with the content from the MR Secret):
```
Warning: spec.template.spec.containers[0].ports[0]: duplicate port definition with spec.template.spec.initContainers[1].ports[0]
```

Reply:
Removed the metrics port from the init container.